internal/sql/pg/conn.go

```diff
@@ -113,9 +121,39 @@ func NewPool(ctx context.Context, cfg *PoolConfig) (*Pool, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	subscribers := syncmap.New[int64, chan<- string]()
+
 	// NOTE: we can consider changing the default exec mode at construction e.g.:
 	// pCfg.ConnConfig.DefaultQueryExecMode = pgx.QueryExecModeSimpleProtocol
 	pCfg.ConnConfig.OnNotice = func(_ *pgconn.PgConn, n *pgconn.Notice) {
+		// Handling the Listen system.
+
+		// if this is the final log for a transaction, send a notice to the subscribers
+		if txid, isFin := isFinishTxid(n.Message); isFin {
+			subscribers.ExclusiveRead(func(m map[int64]chan<- string) {
```

Comment:
I'm doing a more in depth review, but this being `ExclusiveRead` instead of `Exclusive` is probably the reason for the concurrent map write crash.